### PR TITLE
Frontend - Added privacy checkbox and dialog in giver registration summary

### LINKED ISSUE #363: Frontend - Create terms of service in giver registration summary

### PR DESCRIPTION
**Pull request by joffe97**
Fix #363

Initial:
image

Submit when checkbox is not checked:
image

Dialog that opens when `personverserklæring` is pressed:
image
- The checkbox will be checked if the user press `Gi samtykke` in the dialog.

### CODE REVIEW

**Review by TheseusDeus — APPROVED** [review]

File: GiEnJul/ClientApp/src/components/registerGiver/Summary.tsx [hunk 1]
```diff
@@ -274,16 +315,51 @@ const SummaryRegistration: React.FC<Props> = ({
...
+                  Jeg godtar Gi en juls{" "}
+                  <Link
+                    color={state.viewPrivacyError ? "error" : "textSecondary"}
+                    href="#"
+                    onClick={(e) => {
+                      e.preventDefault();
+                      setShowPrivacyDialog(true);
+                    }}
+                  >
+                    personverserklæring
```
Comment (Sommerli) [inline GiEnJul/ClientApp/src/components/registerGiver/Summary.tsx:341]: Typo

**Review by Sommerli — CHANGES_REQUESTED** [review]

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    "Grunnlaget for behandlingen",
+    "Innhenting av personopplysninger",
+    "Utlevering av opplysninger til tredjeparter",
+    "Sletting av personopplysninger",
+    "Rettigheter for den registrerte",
+    "Informasjonssikkerhet",
+  ],
+  text: [
+    "Karianne Munch-Ellingsen er på vegne av Gi en jul behandlingsansvarlig for selskapets behandling av personopplysninger.",
+    "Vi lagrer følgende personopplysninger om våre kunder:navn, adresse, telefonnummer, e-postadresse.",
```
Comment (Sommerli) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:35]:
```suggestion
    "Vi lagrer følgende personopplysninger om våre kunder:navn, telefonnummer, e-postadresse.",
```

File: GiEnJul/ClientApp/src/components/registerGiver/Summary.tsx (hunk 1 above)
Comment (joffe97) [reply GiEnJul/ClientApp/src/components/registerGiver/Summary.tsx:341]: Good catch!

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    "Vi behandler opplysningene for å kunne gjennomføre prosjektet. Personopplysninger vil bli brukt for å koble givere og familier, informasjonsflyt angående prosjektet og generell kontakt om det er nødvendig.",
+    "Informasjon om navn, telefon, e-postadresse benyttes for å oppfylle avtalen. Grunnlaget for denne behandlingen er personvernforordningens artikkel Art 6 (b).",
+    "Vi lagrer de personopplysningene du har avgitt på våre nettsider.Vi bruker informasjonskapsler/cookies på våre nettsider for å gi deg som besøker siden best kundeopplevelse og service.Lov om elektronisk kommunikasjon krever at vi informerer våre besøkende om bruk av informasjonskapsler (cookies). Les mer om bruk av informasjonskapsler her:",
+    "Vi vil ikke dele, selge, overføre eller på annen måte utlevere personopplysninger til andre, med mindre vi er rettslig forpliktet til det.",
+    "Opplysninger vi har mottatt i forbindelse med ditt kjøp lagres i vårt aktive database i 13 måneder.",
```
Comment (Sommerli) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:40]: Kjøp ?

**Review by Sommerli — COMMENTED** [review]
Some text should probably be altered

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx (hunk 2 above)
Comment (TheseusDeus) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:35]: 2. : 
```suggestion
    "Vi lagrer følgende personopplysninger om våre kunder:navn, telefonnummer, e-postadresse.",
```

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    "Sletting av personopplysninger",
+    "Rettigheter for den registrerte",
+    "Informasjonssikkerhet",
+  ],
+  text: [
+    "Karianne Munch-Ellingsen er på vegne av Gi en jul behandlingsansvarlig for selskapets behandling av personopplysninger.",
+    "Vi lagrer følgende personopplysninger om våre kunder:navn, adresse, telefonnummer, e-postadresse.",
+    "Vi behandler opplysningene for å kunne gjennomføre prosjektet. Personopplysninger vil bli brukt for å koble givere og familier, informasjonsflyt angående prosjektet og generell kontakt om det er nødvendig.",
+    "Informasjon om navn, telefon, e-postadresse benyttes for å oppfylle avtalen. Grunnlaget for denne behandlingen er personvernforordningens artikkel Art 6 (b).",
+    "Vi lagrer de personopplysningene du har avgitt på våre nettsider.Vi bruker informasjonskapsler/cookies på våre nettsider for å gi deg som besøker siden best kundeopplevelse og service.Lov om elektronisk kommunikasjon krever at vi informerer våre besøkende om bruk av informasjonskapsler (cookies). Les mer om bruk av informasjonskapsler her:",
```
Comment (TheseusDeus) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:38]:
```suggestion
    "Vi lagrer de personopplysningene du har avgitt på våre nettsider.",
```

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx (hunk 3 above)
Comment (TheseusDeus) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:40]:
```suggestion
   " Opplysninger vi har mottatt i forbindelse med din deltagelse i Gi en jul lagres i vårt aktive database i 13 måneder.",
```

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    "Vi vil ikke dele, selge, overføre eller på annen måte utlevere personopplysninger til andre, med mindre vi er rettslig forpliktet til det.",
+    "Opplysninger vi har mottatt i forbindelse med ditt kjøp lagres i vårt aktive database i 13 måneder.",
+    "Vi behandler dine personopplysninger i henhold til personopplysningsloven og gjeldende forskrifter. Det gjøres oppmerksom på at du kan kreve innsyn i og flytting av egne personopplysninger, samt kreve retting eller sletting av opplysninger. Det kan klages til Datatilsynet på behandling i strid med reglene.",
+    "Vi sikrer dine personopplysninger ved både fysisk og virtuell adgangs- og tilgangskontroll, samt ved kryptering av sensitive deler av avgitte opplysninger.",
```
Comment (TheseusDeus) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:42]:
```suggestion
    "Vi sikrer dine personopplysninger ved best practice for situasjonen",
```

**Review by TheseusDeus — CHANGES_REQUESTED** [review]

**Review by TheseusDeus — DISMISSED** [review]

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx (hunk 5 above)
Comment (Sommerli) [reply GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx]: This definition is somewhat vague. Personal-info should be encrypted as we learnt earlier. Maybe there should be created a issue on the topic(encrypt personal-data - priority high) and then the Personvernerklæring could let the user know that data is encrypted.

File: GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    "Vi behandler opplysningene for å kunne gjennomføre prosjektet. Personopplysninger vil bli brukt for å koble givere og familier, informasjonsflyt angående prosjektet og generell kontakt om det er nødvendig.",
+    "Informasjon om navn, telefon, e-postadresse benyttes for å oppfylle avtalen. Grunnlaget for denne behandlingen er personvernforordningens artikkel Art 6 (b).",
+    "Vi lagrer de personopplysningene du har avgitt på våre nettsider.",
+    "Vi vil ikke dele, selge, overføre eller på annen måte utlevere personopplysninger til andre, med mindre vi er rettslig forpliktet til det.",
+   "Opplysninger vi har mottatt i forbindelse med din deltagelse i Gi en jul lagres i vårt aktive database i 13 måneder.",
+    "Vi behandler dine personopplysninger i henhold til personopplysningsloven og gjeldende forskrifter. Det gjøres oppmerksom på at du kan kreve innsyn i og flytting av egne personopplysninger, samt kreve retting eller sletting av opplysninger. Det kan klages til Datatilsynet på behandling i strid med reglene.",
+    "Vi sikrer dine personopplysninger ved best practice for situasjonen",
```
Comment (joffe97) [inline GiEnJul/ClientApp/src/components/registerGiver/PrivacyDialog.tsx:42]:
```suggestion
    "Vi sikrer dine personopplysninger ved både fysisk og virtuell adgangs- og tilgangskontroll, samt ved kryptering av sensitive deler av avgitte opplysninger.",
```

**Review by Sommerli — APPROVED** [review]

**Pull request merged**